Panic on slice bounds out of range

**Describe the bug**
In the latest 2.10.5 the traffic-manager crashes on start up in some of my environments (mainly inside a [https://www.vcluster.com/](vcluster) with custom cluster domain. It works just fine in 2.10.4. The cluster domain is in the format foo.bar.local

Can't add telepresence client logs as there's nothing to connect to, the traffic manager is installed via helm.

Here's the traffic-manager log:

```
2023-02-08 09:07:53.4509 info    Traffic Manager v2.10.5 [uid:1000,gid:0]
2023-02-08 09:07:53.4518 info    unable to load license: error reading license: open /home/telepresence/license: no such file or directory
2023-02-08 09:07:53.4519 info    starting cloud token watchers
panic: runtime error: slice bounds out of range [23:10]

goroutine 1 [running]:
github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/manager/internal/cluster.NewInfo({0x2e1e478, 0xc0001625d0})
    github.com/telepresenceio/telepresence/v2@v2.10.5/cmd/traffic/cmd/manager/internal/cluster/info.go:100 +0x1eaa
github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/manager.NewService({0x2e1e478?, 0xc000162450?})
    github.com/telepresenceio/telepresence/v2@v2.10.5/cmd/traffic/cmd/manager/service.go:123 +0x3db
github.com/datawire/telepresence-pro/cmd/traffic/manager.NewService({0x2e1e478?, 0xc000162450?})
    github.com/datawire/telepresence-pro/cmd/traffic/manager/service.go:30 +0x27
github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/manager.Main({0x2e1e478, 0xc00023b650}, {0xc00005800a?, 0x4?, 0xc0002b9f70?})
    github.com/telepresenceio/telepresence/v2@v2.10.5/cmd/traffic/cmd/manager/manager.go:75 +0xca3
github.com/datawire/telepresence-pro/cmd/traffic/manager.Main(...)
    github.com/datawire/telepresence-pro/cmd/traffic/manager/main.go:12
main.main()
    github.com/datawire/telepresence-pro/cmd/traffic/main.go:14 +0xf1
Stream closed EOF for ambassador/traffic-manager-6dd9cc77cb-2kmsb (traffic-manager)
```

**Versions:**
traffic-manager: 2.10.5
kubernetes version: v1.25.4+k3s1 

**Related Issues:**
https://github.com/telepresenceio/telepresence/issues/2225
https://github.com/telepresenceio/telepresence/issues/2780

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,8 @@
 - Feature: Adds an authenticator package to support integration with the [client-go credential](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins) plugins when the
   daemon runs in a docker container.
 
+- Bugfix: The traffic-manager will no longer panic when the CNAME of kubernetes.default doesn't contain .svc.
+
 ### 2.11.1 (February 27, 2023)
 
 - Bugfix: The multi-arch build now for the proprietary traffic-manager and traffic-agent now works for both amd64 and arm64.

--- a/cmd/traffic/cmd/manager/internal/cluster/info.go
+++ b/cmd/traffic/cmd/manager/internal/cluster/info.go
@@ -93,11 +93,12 @@ func NewInfo(ctx context.Context) Info {
 
 	apiSvc := "kubernetes.default"
 	var clusterDomain string
-	if cn, err := net.LookupCNAME(apiSvc); err != nil {
+	if cn, err := net.LookupCNAME(apiSvc); err == nil && strings.HasSuffix(cn, apiSvc+".svc.") {
+		clusterDomain = cn[len(apiSvc)+5:]
+	}
+	if clusterDomain == "" {
 		dlog.Infof(ctx, `Unable to determine cluster domain from CNAME of %s: %v"`, err, apiSvc)
 		clusterDomain = "cluster.local."
-	} else {
-		clusterDomain = cn[len(apiSvc)+5:]
 	}
 	dlog.Infof(ctx, "Using cluster domain %q", clusterDomain)
 

--- a/integration_test/itest/cluster.go
+++ b/integration_test/itest/cluster.go
@@ -570,6 +570,7 @@ func (s *cluster) UninstallTrafficManager(ctx context.Context, managerNamespace 
 	// Helm uninstall does deletions asynchronously, so let's wait until the deployment is gone
 	assert.Eventually(t, func() bool { return len(RunningPods(ctx, "traffic-manager", managerNamespace)) == 0 },
 		20*time.Second, 2*time.Second, "traffic-manager deployment was not removed")
+	TelepresenceQuitOk(ctx)
 }
 
 func KubeConfig(ctx context.Context) string {

--- a/pkg/client/rootd/dns/server_darwin.go
+++ b/pkg/client/rootd/dns/server_darwin.go
@@ -155,6 +155,9 @@ func (s *Server) Worker(c context.Context, dev vif.Device, proxyCluster bool, co
 	}
 
 	kubernetesZone := s.clusterDomain
+	if kubernetesZone == "" {
+		kubernetesZone = "cluster.local."
+	}
 	kubernetesZone = kubernetesZone[:len(kubernetesZone)-1] // strip trailing dot
 	rf := resolveFile{
 		port:        dnsAddr.Port,

